src/components/Form/FormProvider.tsx

```diff
@@ -0,0 +1,362 @@
+                        // We delay the validation in order to prevent Checkbox loss of focus when
+                        // the user is focusing a TextInput and proceeds to toggle a CheckBox in
+                        // web and mobile web platforms.
+
+                        setTimeout(() => {
+                            if (
+                                relatedTargetId === CONST.OVERLAY.BOTTOM_BUTTON_NATIVE_ID ||
+                                relatedTargetId === CONST.OVERLAY.TOP_BUTTON_NATIVE_ID ||
+                                relatedTargetId === CONST.BACK_BUTTON_NATIVE_ID
+                            ) {
+                                return;
+                            }
+                            setTouchedInput(inputID);
+                            if (shouldValidateOnBlur) {
+                                onValidate(inputValues, !hasServerError);
+                            }
+                        }, VALIDATE_DELAY);
+                    }
+                    inputProps.onBlur?.(event);
+                },
+                onInputChange: (value: FormValueType, key?: string) => {
+                    const inputKey = key ?? inputID;
+                    setInputValues((prevState) => {
+                        const newState = {
+                            ...prevState,
+                            [inputKey]: value,
+                        };
+
+                        if (shouldValidateOnChange) {
+                            onValidate(newState);
+                        }
+                        return newState as Form;
+                    });
+
+                    if (inputProps.shouldSaveDraft && !(formID as string).includes('Draft')) {
+                        FormActions.setDraftValues(formID as OnyxFormKeyWithoutDraft, {[inputKey]: value});
+                    }
+                    inputProps.onValueChange?.(value, inputKey);
+                },
+            };
+        },
+        [draftValues, formID, errors, formState, hasServerError, inputValues, onValidate, setTouchedInput, shouldValidateOnBlur, shouldValidateOnChange],
+    );
+    const value = useMemo(() => ({registerInput}), [registerInput]);
+
+    return (
+        <FormContext.Provider value={value}>
+            {/* eslint-disable react/jsx-props-no-spreading */}
+            <FormWrapper
+                {...rest}
+                formID={formID}
+                onSubmit={submit}
+                inputRefs={inputRefs}
+                errors={errors}
+                enabledWhenOffline={enabledWhenOffline}
+            >
+                {typeof children === 'function' ? children({inputValues}) : children}
+            </FormWrapper>
+        </FormContext.Provider>
+    );
+}
+
+FormProvider.displayName = 'Form';
+
+export default withOnyx<FormProviderProps, FormProviderOnyxProps>({
+    network: {
```

Comment:
maybe instead of this use `useNetwork`?

Reply:
I don't think is worth it, useNetwork is implemented with withOnyx under the hood, so we would end up with two withOnyx calls.